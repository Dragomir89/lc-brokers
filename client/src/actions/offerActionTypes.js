export const GET_OFFERS = 'GET_OFFERS';
export const POST_OFFER_SUCCESS = 'POST_OFFER_SUCCESS';
export const POST_OFFER_ERROR = 'POST_OFFER_ERROR';
