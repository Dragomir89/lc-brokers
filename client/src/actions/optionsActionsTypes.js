export const GET_OPTIONS = 'GET_OPTIONS';