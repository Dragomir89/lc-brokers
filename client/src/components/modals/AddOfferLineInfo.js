import React from 'react';

const AddOfferLineInfo = ({ label, value, dimension='' }) => {
    const style = !!value ? 'modal__info-position' : 'modal__info-position-danger'
    value = !!value ? `${value} ${dimension}` : '-- няма стойност --';
    
    return (
        <div className={style}>
            <p>{label}: </p>
            <p>{value}</p>
        </div>
    )
}

export default AddOfferLineInfo;