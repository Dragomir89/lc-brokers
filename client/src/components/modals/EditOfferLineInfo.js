import React from 'react'

const EditOfferLineInfo = ({ prevVal, newVal, label }) => {
    const hasChange = prevVal.toString() !== newVal.toString();
    return hasChange ? (
        <div>
            <h3>{label}</h3>
            <p>{prevVal || '-няма-'} ----  {newVal}</p>
        </div>
    ) : (null);
}

export default EditOfferLineInfo;