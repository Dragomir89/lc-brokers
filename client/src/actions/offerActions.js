import { GET_OFFERS, POST_OFFER_SUCCESS, POST_OFFER_ERROR, GET_CURRENT_OFFER } from './offerActionTypes';
import axios from 'axios';

export const getOffers = (page, queryStr) => async (dispath) => {
    const res = await axios.get(`http://localhost:5000/api/get-offers/${page}?${queryStr}`);
    return dispath({ type: GET_OFFERS, payload: res.data.offers });
}

export const postOffer = (data) => async (dispatch) => {
    data.addedOn = new Date();
    const options = { method: 'post', url: 'http://localhost:5000/api/offer', data };
    const res = await axios(options);
    const type = !!res.data.error ? POST_OFFER_ERROR : POST_OFFER_SUCCESS
    dispatch({ type });
}

export const updateOffer = (data) => async (dispatch) => {
    data.addedOn = new Date();
    const options = { method: 'post', url: `http://localhost:5000/api/offer/${data._id}`, data };
    const res = await axios(options);
    const type = !!res.data.error ? POST_OFFER_ERROR : POST_OFFER_SUCCESS
    dispatch({ type, payload: res.data });
}

export const getCurrentOffer = (id) => async dispatch => {
    const res = await axios.get(`http://localhost:5000/api/offer/${id}`);
    dispatch({type: GET_CURRENT_OFFER, payload: res.data});
}