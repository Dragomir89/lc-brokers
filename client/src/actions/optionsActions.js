import { GET_OPTIONS } from './optionsActionsTypes';

import axios from 'axios';

export const getOptions = () => async (dispath) => {
    const res = await axios.get('http://localhost:5000/api/options');
    return dispath({ type: GET_OPTIONS, payload: res.data });
}
