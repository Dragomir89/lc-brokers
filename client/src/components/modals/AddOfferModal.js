import React from 'react';
import { connect } from 'react-redux';
import Modal from 'react-modal';
import AddOfferLineInfo from './AddOfferLineInfo';
import { POST_OFFER_ERROR } from '../../actions/offerActionTypes';
import { postOffer } from '../../actions/offerActions';

const AddOfferModal = ({ isOpen, offerVlues, closeHandlerModal, postOffer }) => {

    const {
        constructionType,
        propertyType,
        stateVal,
        neighborhood,
        price,
        area,
        description,
        floor,
        number,
        address,
        propertyOwnerName,
        phoneNumber,
        phoneNumber2,
        phoneNumber3,
        lastCall,
        nextCall,
        offerPostStatus } = offerVlues;

    const showReqMessage = () => {
        let message = ''
        if (offerPostStatus) {
            message = offerPostStatus === POST_OFFER_ERROR ?
                (<p className='messages__fail-request'>! Офертата не беше качена !</p>) :
                (<p className='messages__success-request'>Добавихте офертата успешно</p>);
        }
        return message;
    }

    const saveOfferHandler = () => { /// REMOVE !!!!!
        const { nextCall, lastCall } = offerVlues;
        postOffer({ ...offerVlues, lastCall: lastCall.toISOString(), nextCall: nextCall.toISOString() });
    }

    return (
        <Modal
            ariaHideApp={false}
            isOpen={isOpen}
            contentLabel='first modal'
            onRequestClose={closeHandlerModal}
            closeTimeoutMS={500}
            className='modal'
        >
            <h1 className='title__offer title__center'>Искате ли да запазите тази оферта ?</h1>
            <div className='model__test-style content-container__offer-modal'>

                <AddOfferLineInfo label='Строителство' value={constructionType} />
                <AddOfferLineInfo label='Вид Имот' value={propertyType} />
                <AddOfferLineInfo label='Състояние' value={stateVal} />
                <AddOfferLineInfo label='Квартал' value={neighborhood} />

                <AddOfferLineInfo label='Цена' value={price} dimension='евро' />
                <AddOfferLineInfo label='Квадратура' value={area} dimension='кв/м' />
                <AddOfferLineInfo label='Описание' value={description} />
                <AddOfferLineInfo label='Номер На Оферта' value={number} />
                <AddOfferLineInfo label='Адрес' value={address} />
                <AddOfferLineInfo label='Етаж' value={floor} dimension={' ет'} />
                <AddOfferLineInfo label='Собственик' value={propertyOwnerName} />

                <AddOfferLineInfo label='Главен Тлефон' value={phoneNumber} />
                <AddOfferLineInfo label='Тлефон 2' value={phoneNumber2} />
                <AddOfferLineInfo label='Тлефон 3' value={phoneNumber3} />
                <AddOfferLineInfo label='Последно обаждане' value={lastCall.format("MM/DD/YYYY")} />
                <AddOfferLineInfo label='Следващо обаждане' value={nextCall.format("MM/DD/YYYY")} />


                <div>
                    {showReqMessage()}
                </div>


                <div className='button__relative-wrapper'>
                    <button
                        onClick={closeHandlerModal}
                        className='button__success'>
                        Назад
                </button>
                    <button
                        onClick={saveOfferHandler}
                        className='button__danger button__offer-details-position'>
                        Запази офертата !
                    </button>
                </div>
            </div>

        </Modal>
    )

}

const actions = { postOffer };

export default connect(null, actions)(AddOfferModal);
