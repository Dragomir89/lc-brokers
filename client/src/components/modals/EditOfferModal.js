import React from 'react';
import { connect } from 'react-redux';
import { updateOffer } from '../../actions/offerActions';
import Modal from 'react-modal';
import EditOfferLineInfo from './EditOfferLineInfo';

const EditOfferModal = ({ isOpen, offerVlues, closeHandlerModal, options, editOffer, updateOffer }) => {

    const {
        constructionTypeId,
        constructionType,
        propertyTypeId,
        propertyType,
        state,
        stateVal,
        neighborhoodId,
        neighborhood,
        price,
        area,
        description,
        floor,
        address,
        propertyOwnerName,
        phoneNumber,
        phoneNumber2,
        phoneNumber3,
        lastCall,
        nextCall,
    } = offerVlues;

    const checkSelectChange = (selectType, label, oldPropId, newPropId, newProp) => {
        if (newPropId !== oldPropId && (options[selectType] && options[selectType].length > 0)) {
            const option = options[selectType].find(e => e._id === oldPropId);
            return option && (<EditOfferLineInfo label={label} prevVal={option.value} newVal={newProp} />)
        }
    }

    const saveEditedOffer = (e)=> {
        e.preventDefault();
        updateOffer({
            ...offerVlues,
            lastCall: lastCall.toISOString(), 
            nextCall: nextCall.toISOString() 
        });
    }
    
    console.log('render')
    return (
        <Modal
            ariaHideApp={false}
            isOpen={isOpen}
            contentLabel='second modal'
            onRequestClose={closeHandlerModal}
            closeTimeoutMS={500}
            className='modal'
        >
            <h1 className='title__offer title__center'>Искате ли да запазите промените ?</h1>
            <div className='model__test-style content-container__offer-modal'>

                {checkSelectChange('constructionTypes',
                    'Вид стротелство',
                    editOffer.constructionTypeId,
                    constructionTypeId,
                    constructionType)}
                {checkSelectChange('propertyTypes',
                    'Вид имот',
                    editOffer.propertyTypeId,
                    propertyTypeId,
                    propertyType)}
                {checkSelectChange('states',
                    'Състояние',
                    editOffer.state,
                    state,
                    stateVal)}
                {checkSelectChange('neighborhoods',
                    'Квартал',
                    editOffer.neighborhoodId,
                    neighborhoodId,
                    neighborhood
                )}

                <EditOfferLineInfo label='Площ' prevVal={editOffer.price} newVal={price} />
                <EditOfferLineInfo label='Етаж' prevVal={editOffer.floor} newVal={floor} />
                <EditOfferLineInfo label='Площ' prevVal={editOffer.area} newVal={area} />
                <EditOfferLineInfo label='Описание' prevVal={editOffer.description} newVal={description} />
                <EditOfferLineInfo label='Адрес' prevVal={editOffer.address} newVal={address} />
                <EditOfferLineInfo label='Собственик' prevVal={editOffer.propertyOwnerName} newVal={propertyOwnerName} />
                <EditOfferLineInfo label='Тлефон Главен' prevVal={editOffer.phoneNumber} newVal={phoneNumber} />
                <EditOfferLineInfo label='Тлефон 2' prevVal={editOffer.phoneNumber2} newVal={phoneNumber2} />
                <EditOfferLineInfo label='Тлефон 3' prevVal={editOffer.phoneNumber3} newVal={phoneNumber3} />
                <EditOfferLineInfo label='Предишно обаждане'
                    prevVal={editOffer.lastCall && editOffer.lastCall.format('DD/MM/YYYY')}
                    newVal={lastCall && lastCall.format('DD/MM/YYYY')} />
                <EditOfferLineInfo label='Следващо обаждане'
                    prevVal={editOffer.nextCall && editOffer.nextCall.format('DD/MM/YYYY')}
                    newVal={nextCall && nextCall.format('DD/MM/YYYY')} />
                <div style={
                    {
                        display: 'flex',
                        justifyContent: 'space-between'
                    }}>
                    <button
                        className='button__success'
                        onClick={closeHandlerModal}>
                        Назад
                    </button>
                    <button className='button__danger'>
                        Изтриване
                    </button>
                    <button
                        className='button__primary'
                        onClick={saveEditedOffer}
                        >
                        Запази
                    </button>
                </div>
            </div>
        </Modal>
    )

}

const mapStateToProps = (state) => {
    return {
        editOffer: state.editOffer,
        options: state.options
    }
}

const actions = {
    updateOffer
}

export default connect(mapStateToProps, actions)(EditOfferModal);
