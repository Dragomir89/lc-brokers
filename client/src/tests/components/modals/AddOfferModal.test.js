import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import '../../setupTests';
import { AddOfferModal } from '../../../components/modals/AddOfferModal';
import { addModalData } from '../../test-data';
import { POST_OFFER_ERROR } from '../../../actions/offerActionTypes';
import moment from 'moment';

test('should call postOffer with correct values', () => {
    const postOfferSpy = jest.fn();
    const lastCall = moment(addModalData.lastCall);
    const nextCall = moment(addModalData.nextCall);
    const values = {...addModalData, lastCall, nextCall}
    const wrapper = shallow(<AddOfferModal postOffer={postOfferSpy} offerVlues={values}/>)
    expect(toJSON(wrapper)).toMatchSnapshot();
    wrapper.find('button').at(1).simulate('click');
    const postObject = { 
        ...values, 
        lastCall: values.lastCall.toISOString(), 
        nextCall: values.nextCall.toISOString() 
    }
    expect(postOfferSpy).toHaveBeenLastCalledWith(postObject);
});

test('should call postOffer with error message values', () => {
    const postOfferSpy = jest.fn();
    const lastCall = moment(addModalData.lastCall);
    const nextCall = moment(addModalData.nextCall);
    const values = {...addModalData, lastCall, nextCall, offerPostStatus: POST_OFFER_ERROR}
    const wrapper = shallow(<AddOfferModal postOffer={postOfferSpy} offerVlues={values}/>)
    expect(toJSON(wrapper)).toMatchSnapshot();
    
});