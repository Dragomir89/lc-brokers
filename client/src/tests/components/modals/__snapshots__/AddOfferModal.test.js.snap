// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should call postOffer with correct values 1`] = `
<Modal
  ariaHideApp={false}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal"
  closeTimeoutMS={500}
  contentLabel="first modal"
  isOpen={false}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <h1
    className="title__offer title__center"
  >
    Искате ли да запазите тази оферта ?
  </h1>
  <div
    className="model__test-style content-container__offer-modal"
  >
    <AddOfferLineInfo
      label="Строителство"
      value="НОВО"
    />
    <AddOfferLineInfo
      label="Вид Имот"
      value="ЧЕТЕРИСТАЕН"
    />
    <AddOfferLineInfo
      label="Състояние"
      value="НАЕТ"
    />
    <AddOfferLineInfo
      label="Квартал"
      value="ЛОЗЕНЕЦ"
    />
    <AddOfferLineInfo
      dimension="евро"
      label="Цена"
      value={400}
    />
    <AddOfferLineInfo
      dimension="кв/м"
      label="Квадратура"
      value={40}
    />
    <AddOfferLineInfo
      label="Описание"
      value=""
    />
    <AddOfferLineInfo
      label="Номер На Оферта"
      value={1}
    />
    <AddOfferLineInfo
      label="Адрес"
      value="UPDATE_0001"
    />
    <AddOfferLineInfo
      dimension=" ет"
      label="Етаж"
      value={6}
    />
    <AddOfferLineInfo
      label="Собственик"
      value="тест"
    />
    <AddOfferLineInfo
      label="Главен Тлефон"
      value="1111"
    />
    <AddOfferLineInfo
      label="Тлефон 2"
    />
    <AddOfferLineInfo
      label="Тлефон 3"
    />
    <AddOfferLineInfo
      label="Последно обаждане"
      value="10/09/2018"
    />
    <AddOfferLineInfo
      label="Следващо обаждане"
      value="10/09/2018"
    />
    <div />
    <div
      className="button__relative-wrapper"
    >
      <button
        className="button__success"
      >
        Назад
      </button>
      <button
        className="button__danger button__offer-details-position"
        onClick={[Function]}
      >
        Запази офертата !
      </button>
    </div>
  </div>
</Modal>
`;

exports[`should call postOffer with error message values 1`] = `
<Modal
  ariaHideApp={false}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal"
  closeTimeoutMS={500}
  contentLabel="first modal"
  isOpen={false}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <h1
    className="title__offer title__center"
  >
    Искате ли да запазите тази оферта ?
  </h1>
  <div
    className="model__test-style content-container__offer-modal"
  >
    <AddOfferLineInfo
      label="Строителство"
      value="НОВО"
    />
    <AddOfferLineInfo
      label="Вид Имот"
      value="ЧЕТЕРИСТАЕН"
    />
    <AddOfferLineInfo
      label="Състояние"
      value="НАЕТ"
    />
    <AddOfferLineInfo
      label="Квартал"
      value="ЛОЗЕНЕЦ"
    />
    <AddOfferLineInfo
      dimension="евро"
      label="Цена"
      value={400}
    />
    <AddOfferLineInfo
      dimension="кв/м"
      label="Квадратура"
      value={40}
    />
    <AddOfferLineInfo
      label="Описание"
      value=""
    />
    <AddOfferLineInfo
      label="Номер На Оферта"
      value={1}
    />
    <AddOfferLineInfo
      label="Адрес"
      value="UPDATE_0001"
    />
    <AddOfferLineInfo
      dimension=" ет"
      label="Етаж"
      value={6}
    />
    <AddOfferLineInfo
      label="Собственик"
      value="тест"
    />
    <AddOfferLineInfo
      label="Главен Тлефон"
      value="1111"
    />
    <AddOfferLineInfo
      label="Тлефон 2"
    />
    <AddOfferLineInfo
      label="Тлефон 3"
    />
    <AddOfferLineInfo
      label="Последно обаждане"
      value="10/09/2018"
    />
    <AddOfferLineInfo
      label="Следващо обаждане"
      value="10/09/2018"
    />
    <div>
      <p
        className="messages__fail-request"
      >
        ! Офертата не беше качена !
      </p>
    </div>
    <div
      className="button__relative-wrapper"
    >
      <button
        className="button__success"
      >
        Назад
      </button>
      <button
        className="button__danger button__offer-details-position"
        onClick={[Function]}
      >
        Запази офертата !
      </button>
    </div>
  </div>
</Modal>
`;
